rllab/dynamics_randomization/mujoco_model_gen.py

```diff
@@ -0,0 +1,127 @@
+import atexit
+from queue import Queue
+from threading import Event
+from threading import Thread
```

Comment:
It will be important to benchmark this to ensure threads (as opposed to processes) actually speed things up.

Python has a design detail called the GIL (global interpreter lock), which means that usually only one thread inside a Python application may run at once. This does not necessarily apply to parts of Python programs which are implemented using native code (e.g. mujoco), but whether that code is blocked by the GIL is very implementation-specific.